website_sale_qty/static/src/js/website_sale_qty.js

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html) */
+
+odoo.define('website_sale_qty', function(require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+    base.ready().done(function() {
+        var radios = $('.js-pqt-input');
+
+        if (radios.length > 0) {
+            var radioMap = {};
+            _.each(radios, function(element) {
+                var $element = $(element);
+                radioMap[$element.val()] = $element;
+            });
+            var quantityInput = $('.css_quantity input');
+            var quantityToggles = $('.css_quantity a');
+
+            radios.click(function(event) {
+                var newValue = $(event.currentTarget).val();
+                quantityInput.val(newValue).trigger('change');
+            });
+
+            var updateRadios = function() {
+                radios.prop('checked', false);
+
+                var newValue = quantityInput.val();
+                if (newValue in radioMap) {
+                    radioMap[newValue].prop('checked', true);
+                }
+            };
+
+            quantityInput.on('input', updateRadios);
+            quantityToggles.click(function() {
+                // This allows website_sale.website_sale to update the input first
```

Comment:
Fair enough. I was thinking of a race condition as a scenario where multiple threads or processes are interacting with the same memory in parallel. This doesn't happen in JS, where callbacks are atomic and get executed sequentially, but if the sequence is not predictable, you basically run into the same problem of memory being in an unexpected state, so in this case it's mostly just semantics.

I'll expand on the comment a bit.